File: Phan_A/baitap35/baitap35.py
import sys

def sang(n: int) -> list:
    if n < 2:
        return []
    check = [True] * (n + 1)
    check[0] = check[1] = False
    p = 2
    while p * p <= n:
        if check[p]:
            for i in range(p * p, n + 1, p):
                check[i] = False
        p += 1
    return [p for p in range(2, n + 1) if check[p]]

def main():
    content = sys.stdin.read().strip()
    if not content:
        return
    
    tokens = content.split()
    for token in tokens:
        n = int(token)
        primes = sang(n)
        if n <= 30:
            print(" ".join(map(str, primes)) + f" ({len(primes)} so)", flush=True)
        elif n == 1000000:
            tong = sum(primes)
            print(f"so luong = {len(primes)}, tong = {tong}", flush=True)
        else:
            print(f"so luong = {len(primes)}", flush=True)

if __name__ == "__main__":
    main()

File: Phan_A/baitap35/baitap35.jl
function la_nguyen_to(n::Int)
    if n < 2
        return false
    end
    i = 2
    while i * i <= n
        if n % i == 0
            return false
        end
        i += 1
    end
    return true
end

function sang(n::Int)
    check = trues(n)
    if n >= 1
        check[1] = false
    end
    p = 2
    while p * p <= n
        if check[p]
            for i in (p * p):p:n
                check[i] = false
            end
        end
        p += 1
    end
    primes = Int[]
    for i in 1:n
        if check[i]
            push!(primes, i)
        end
    end
    return primes
end

function main()
    for line in eachline(stdin)
        line = strip(line)
        if isempty(line)
            continue
        end
        n = parse(Int, line)
        primes = sang(n)
        if n <= 30
            println(join(primes, " ") * " ($(length(primes)) so)")
        else
            tong = sum(primes)
            if n == 1000000
                println("so luong = $(length(primes)), tong = $(tong)")
            else
                println("so luong = $(length(primes))")
            end
        end
    end
end

main()